src/object/obj_tx.c

```diff
@@ -3273,7 +3285,8 @@ dc_tx_convert(struct dc_object *obj, enum obj_rpc_opc opc, tse_task_t *task)
 	int		 rc = 0;
 
 	D_ASSERT(obj != NULL);
-	D_ASSERTF(task->dt_result == 0, "Unexpected initial task result %d\n", task->dt_result);
+	D_ASSERTF(task->dt_result == 0 || task->dt_result == -DER_NEED_TX,
```

Comment:
sorry, just confirm, if the original test get -DER_NEED_TX then it will be rescheduled and in task_reinit() it will reset the task->dt_result as 0. Could you please give a detailed executing path can get -DER_NEED_TX here? thanks

Reply:
The patch has been refreshed without the re-sched.